Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/EditorAction.cpp

```diff
@@ -55,26 +57,83 @@ namespace AzToolsFramework
         }
     }
 
+    const AZStd::string& EditorAction::GetName()
+    {
+        return m_name;
+    }
+
+    void EditorAction::SetName(AZStd::string name)
+    {
+        m_name = AZStd::move(name);
+    }
+
+    const AZStd::string& EditorAction::GetDescription()
+    {
+        return m_description;
+    }
+
+    void EditorAction::SetDescription(AZStd::string description)
+    {
+        m_description = AZStd::move(description);
+    }
+
+    const AZStd::string& EditorAction::GetCategory()
+    {
+        return m_category;
+    }
+
+    void EditorAction::SetCategory(AZStd::string category)
+    {
+        m_category = AZStd::move(category);
+    }
+
+    const AZStd::string& EditorAction::GetIconPath()
+    {
+        return m_iconPath;
+    }
+
+    void EditorAction::SetIconPath(AZStd::string iconPath)
+    {
+        m_iconPath = AZStd::move(iconPath);
+        RetrieveIconFromPath();
+
+        if (!m_icon.isNull())
+        {
+            m_action->setIcon(m_icon);
+        }
+    }
+
     QAction* EditorAction::GetAction()
     {
         // Update the action to ensure it is visualized correctly.
         Update();
 
         return m_action;
     }
-    
+
     void EditorAction::Update()
     {
-        if (m_updateCallback)
+        if (m_checkStateCallback)
         {
             // Refresh checkable action value.
-            m_action->setChecked(m_updateCallback());
+            m_action->setChecked(m_checkStateCallback());
         }
     }
-    
+
     bool EditorAction::IsCheckable()
     {
         return m_action->isCheckable();
     }
 
+    void EditorAction::RetrieveIconFromPath()
```

Comment:
suggestion: `Refresh` might be clearer than `Retrieve` here as we're not returning a value